charts/prometheus-thanos/templates/receiver/service.yaml

```diff
@@ -23,6 +23,10 @@ spec:
       targetPort: http
       protocol: TCP
       name: http
+    - port: {{ .Values.service.receiver.httpRemoteWrite.port }}
+      targetPort: http-rw
```

Comment:
@tdinucci May I don't get the point here. What is the mapping for and why is there no target port inside the deployment?

Reply:
Sorry if I'm misinterpreting your question.

This port should relate to: https://github.com/kiwigrid/helm-charts/blob/ef2cdc0c52921f87e036cf26d5aaaaf26168f2c4/charts/prometheus-thanos/templates/receiver/statefulset.yaml#L90

Since this is a headless service, ports obviously can't be mapped as per normal services - meaning as far as k8s is concerned that the port number here is useless since connections are simply redirected to pods with the callers requested port number.  This is why the same Helm variable is referenced here as in: https://github.com/kiwigrid/helm-charts/blob/ef2cdc0c52921f87e036cf26d5aaaaf26168f2c4/charts/prometheus-thanos/templates/receiver/statefulset.yaml#L91

Certain systems though (in our case Istio) may depend on there being a port definition on the service though and this is the reason for this PR.